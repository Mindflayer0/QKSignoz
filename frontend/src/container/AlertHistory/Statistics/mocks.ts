/* eslint-disable sonarjs/no-duplicate-string */
export const statsData = {
	totalCurrentTriggers: 14,
	totalPastTriggers: 15,
	currentTriggersSeries: {
		values: [
			{
				timestamp: '1625097600000', // Sample epoch value
				value: 100,
			},
		],
	},
	pastTriggersSeries: {
		values: [
			{
				timestamp: '1625097600000', // Sample epoch value
				value: 100,
			},
		],
	},
	currentAvgResolutionTime: 1.3,
	pastAvgResolutionTime: 3.2,
	currentAvgResolutionTimeSeries: {
		values: [
			{
				timestamp: '1625097600000', // Sample epoch value
				value: 100,
			},
		],
	},
	pastAvgResolutionTimeSeries: {
		values: [
			{
				timestamp: '1625097600000', // Sample epoch value
				value: 100,
			},
		],
	},
};

export const topContributorsData = {
	contributors: [
		{
			labels: {
				operation: 'POST /transaction-module/record-payment',
				service_name: 'order-service-prod',
				k3: 'v3',
			},
			count: 6,
		},
		{
			labels: {
				operation: 'GET /financial-module/account-statement',
				service_name: 'catalog-manager-001',
				k3: 'v3',
			},
			count: 4,
		},
		{
			labels: {
				operation: 'GET /financial-module/account-statement',
				service_name: 'catalog-manager-001',
				k3: 'v3',
			},
			count: 2,
		},
		{
			labels: {
				operation: 'GET /financial-module/account-statement',
				service_name: 'catalog-manager-001',
				k3: 'v3',
			},
			count: 2,
		},
	],
};
