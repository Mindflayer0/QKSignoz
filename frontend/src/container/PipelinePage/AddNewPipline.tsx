import { PlusOutlined } from '@ant-design/icons';
import TextToolTip from 'components/TextToolTip';
import React from 'react';
import { useTranslation } from 'react-i18next';

import { Button, ButtonContainer } from './styles';

function AddNewPipline({ setActionType }: AddNewPiplineType): JSX.Element {
	const { t } = useTranslation(['pipeline']);

	return (
		<ButtonContainer>
			<TextToolTip text="Add Piplines" />
			<Button
				icon={<PlusOutlined />}
				onClick={(): void => setActionType('add-pipeline')}
				type="primary"
			>
				{t('new_pipeline')}
			</Button>
		</ButtonContainer>
	);
}

interface AddNewPiplineType {
	setActionType: (b?: string) => void;
}

export default AddNewPipline;
