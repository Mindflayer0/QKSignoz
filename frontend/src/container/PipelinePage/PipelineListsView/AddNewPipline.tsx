import { Button, Divider, Form, Input, Modal, Typography } from 'antd';
import type { FormInstance } from 'antd/es/form';
import React, { RefObject, useEffect, useMemo, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { v4 as uuid } from 'uuid';

import TagInput from '../components/TagInput';
import { ActionType } from '../Layouts';
import PiplinesSearchSection from '../Layouts/PiplinesSearchSection';
import { PipelineColumn } from '.';
import { addPipelinefieldLists } from './config';
import { FormLabelStyle } from './styles';

function AddNewPipline({
	isActionType,
	setActionType,
	selectedRecord,
	setDataSource,
	formRef,
	handleModalCancelAction,
	dataSource,
}: AddNewPiplineProps): JSX.Element {
	const [form] = Form.useForm();
	const { t } = useTranslation(['pipeline', 'common']);
	const [count, setCount] = useState(3);
	const [tagsListData, setTagsListData] = useState<PipelineColumn['tags']>();

	const isEdit = useMemo(() => isActionType === 'edit-pipeline', [isActionType]);
	const isAdd = useMemo(() => isActionType === 'add-pipeline', [isActionType]);

	useEffect(() => {
		if (isEdit) {
			form.setFieldsValue({
				name: selectedRecord?.name,
				tags: selectedRecord?.tags,
			});
			setTagsListData(selectedRecord?.tags);
		} else {
			setTagsListData([]);
		}
	}, [form, isEdit, selectedRecord?.name, selectedRecord?.tags]);

	const onFinish = (values: PipelineColumn): void => {
		const operatorsData = Array({
			name: values.operators,
		});

		const newData = {
			orderid: count.toString(),
			key: uuid(),
			editedBy: '',
			filter: '',
			lastEdited: new Date().toDateString(),
			name: values.name,
			tags: tagsListData,
			operators: operatorsData,
		};

		if (isEdit) {
			const findRecordIndex = dataSource.findIndex(
				(i) => i.name === selectedRecord?.name,
			);
			const updatedPipelineData = {
				...dataSource[findRecordIndex],
				name: values.name,
				tags: tagsListData,
			};

			const tempData = dataSource?.map((data) =>
				data.name === selectedRecord?.name ? updatedPipelineData : data,
			);
			setDataSource(tempData as Array<PipelineColumn>);
			formRef?.current?.resetFields();
		} else {
			setTagsListData([]);
			setCount((prevState: number) => (prevState + 1) as number);
			setDataSource(
				(pre: PipelineColumn[]) => [...pre, newData] as PipelineColumn[],
			);
			formRef?.current?.resetFields();
		}
		setActionType(undefined);
	};

	const onCancelHandler = (): void => {
		setActionType(undefined);
		formRef?.current?.resetFields();
	};

	return (
		<Modal
			title={
				<Typography.Title
					level={4}
					style={{
						fontStyle: 'normal',
						fontWeight: 600,
						fontSize: '1.125rem',
						lineHeight: '1.5rem',
					}}
				>
					{isEdit
						? `${t('edit_pipeline')} : ${selectedRecord?.name}`
						: t('create_pipeline')}
				</Typography.Title>
			}
			centered
			open={isEdit || isAdd}
			width={800}
			footer={null}
			onCancel={onCancelHandler}
		>
			<Divider plain />
			<div style={{ marginTop: '1.563rem' }}>
				<span>{t('filter')}</span>
				<div style={{ marginTop: '0.313rem' }}>
					<PiplinesSearchSection />
				</div>
			</div>
			<Form
				form={form}
				layout="vertical"
				style={{ marginTop: '1.25rem' }}
				onFinish={onFinish}
				ref={formRef}
			>
				{addPipelinefieldLists.map((i) => {
					if (i.id === 3) {
						return (
							<Form.Item
								required={false}
								name={i.name}
								label={i.fieldName}
								key={i.id}
								rules={[
									{
										required: true,
									},
								]}
							>
								<Input.TextArea
									rows={3}
									name={i.fieldName}
									placeholder={
										isEdit
											? `This is a pipeline to edit ${selectedRecord?.name}`
											: i.placeholder
									}
								/>
							</Form.Item>
						);
					}
					if (i.id === 2) {
						return (
							<Form.Item
								required={false}
								label={<FormLabelStyle>{i.fieldName}</FormLabelStyle>}
								key={i.id}
								name={i.name}
							>
								<TagInput
									setTagsListData={setTagsListData}
									tagsListData={tagsListData as []}
									placeHolder={i.fieldName}
								/>
							</Form.Item>
						);
					}
					return (
						<Form.Item
							required={false}
							label={<FormLabelStyle>{i.fieldName}</FormLabelStyle>}
							key={i.id}
							rules={[
								{
									required: true,
								},
							]}
							name={i.name}
						>
							<Input name={i.fieldName} placeholder={i.placeholder} />
						</Form.Item>
					);
				})}
				<Divider plain />
				<Form.Item>
					<div
						style={{ display: 'flex', flexDirection: 'row-reverse', gap: '0.625rem' }}
					>
						<Button key="submit" type="primary" htmlType="submit">
							{isEdit ? t('update') : t('create')}
						</Button>
						<Button key="cancel" onClick={handleModalCancelAction}>
							{t('cancel')}
						</Button>
					</div>
				</Form.Item>
			</Form>
		</Modal>
	);
}

interface AddNewPiplineProps {
	isActionType: string;
	setActionType: (actionType?: ActionType) => void;
	selectedRecord: PipelineColumn | undefined;
	setDataSource: (value: React.SetStateAction<Array<PipelineColumn>>) => void;
	formRef: RefObject<FormInstance>;
	handleModalCancelAction: VoidFunction;
	dataSource: Array<PipelineColumn>;
}

export default AddNewPipline;
