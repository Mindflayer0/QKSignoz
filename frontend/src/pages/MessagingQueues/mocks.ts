/* eslint-disable sonarjs/no-duplicate-string */
export const consumerGrpResponse = {
	status: 'success',
	data: {
		resultType: '',
		result: [
			{
				table: {
					columns: [
						{
							name: 'service_name',
							queryName: '',
							isValueColumn: false,
						},
						{
							name: 'p99',
							queryName: '',
							isValueColumn: false,
						},
						{
							name: 'error_rate',
							queryName: '',
							isValueColumn: false,
						},
						{
							name: 'throughput',
							queryName: '',
							isValueColumn: false,
						},
						{
							name: 'avg_msg_size',
							queryName: '',
							isValueColumn: false,
						},
					],
					rows: [
						{
							data: {
								avg_msg_size: '0',
								error_rate: '0',
								p99: '0.2942205100000016',
								service_name:
									'consumer-svc-consumer-svc-consumer-svc-consumer-svc-consumer-svc',
								throughput: '0.00016534391534391533',
							},
						},
						{
							data: {
								avg_msg_size: '1',
								error_rate: '2',
								p99: '0.294',
								service_name: 'frontend',
								throughput: '0.00016534391534391533',
							},
						},
						{
							data: {
								avg_msg_size: '0',
								error_rate: '0',
								p99: '0.2942205100000016',
								service_name: 'consumer-svc',
								throughput: '0.00016534391534391533',
							},
						},
						{
							data: {
								avg_msg_size: '0',
								error_rate: '0',
								p99: '0.2942205100000016',
								service_name: 'consumer-svc',
								throughput: '0.00016534391534391533',
							},
						},
						{
							data: {
								avg_msg_size: '0',
								error_rate: '0',
								p99: '0.2942205100000016',
								service_name: 'consumer-svc',
								throughput: '0.00016534391534391533',
							},
						},
						{
							data: {
								avg_msg_size: '0',
								error_rate: '0',
								p99: '0.2942205100000016',
								service_name: 'consumer-svc',
								throughput: '0.00016534391534391533',
							},
						},
						{
							data: {
								avg_msg_size: '1',
								error_rate: '2',
								p99: '0.294',
								service_name: 'frontend',
								throughput: '0.00016534391534391533',
							},
						},
						{
							data: {
								avg_msg_size: '0',
								error_rate: '0',
								p99: '0.2942205100000016',
								service_name: 'consumer-svc',
								throughput: '0.00016534391534391533',
							},
						},
						{
							data: {
								avg_msg_size: '0',
								error_rate: '0',
								p99: '0.2942205100000016',
								service_name: 'consumer-svc',
								throughput: '0.00016534391534391533',
							},
						},
						{
							data: {
								avg_msg_size: '0',
								error_rate: '0',
								p99: '0.2942205100000016',
								service_name: 'consumer-svc',
								throughput: '0.00016534391534391533',
							},
						},
					],
				},
			},
		],
	},
};

export const productDetailResponse = {
	status: 'success',
	data: {
		resultType: '',
		result: [
			{
				table: {
					columns: [
						{
							name: 'service_name',
							queryName: '',
							isValueColumn: false,
						},
						{
							name: 'p99_query.p99',
							queryName: '',
							isValueColumn: false,
						},
						{
							name: 'error_rate',
							queryName: '',
							isValueColumn: false,
						},
						{
							name: 'rps',
							queryName: '',
							isValueColumn: false,
						},
					],
					rows: [
						{
							data: {
								error_rate: '0',
								'p99_query.p99': '150.08830908000002',
								rps: '0.00016534391534391533',
								service_name: 'producer-svc',
							},
						},
					],
				},
			},
		],
	},
};

export const networLatencyResponse = {
	status: 'success',
	data: {
		resultType: '',
		result: [
			{
				table: {
					columns: [
						{
							name: 'service_name',
							queryName: '',
							isValueColumn: false,
						},
						{
							name: 'service_instance_id',
							queryName: '',
							isValueColumn: false,
						},
						{
							name: 'latency_0',
							queryName: 'latency_0',
							isValueColumn: true,
						},
						{
							name: 'latency_1',
							queryName: 'latency_1',
							isValueColumn: true,
						},
						{
							name: 'latency_2',
							queryName: 'latency_2',
							isValueColumn: true,
						},
						{
							name: 'latency_3',
							queryName: 'latency_3',
							isValueColumn: true,
						},
					],
					rows: [
						{
							data: {
								latency_0: 3230.1,
								latency_1: 'n/a',
								latency_2: 'n/a',
								latency_3: 'n/a',
								service_instance_id: 'a96ff029-6f14-435a-a3d4-ab4742b4347f',
								service_name: 'consumer-svc',
							},
						},
						{
							data: {
								latency_0: 503,
								latency_1: 'n/a',
								latency_2: 'n/a',
								latency_3: 'n/a',
								service_instance_id: 'e33ffd7c-827a-427a-828e-547e00cb80d8',
								service_name: 'consumer-svc',
							},
						},
						{
							data: {
								latency_0: 502.62,
								latency_1: 'n/a',
								latency_2: 'n/a',
								latency_3: 'n/a',
								service_instance_id: '9e87227f-a564-4b55-bf7c-fb00365d9400',
								service_name: 'consumer-svc',
							},
						},
						{
							data: {
								latency_0: 'n/a',
								latency_1: 3230.1,
								latency_2: 'n/a',
								latency_3: 'n/a',
								service_instance_id: 'a96ff029-6f14-435a-a3d4-ab4742b4347f',
								service_name: 'consumer-svc',
							},
						},
						{
							data: {
								latency_0: 'n/a',
								latency_1: 503,
								latency_2: 'n/a',
								latency_3: 'n/a',
								service_instance_id: 'e33ffd7c-827a-427a-828e-547e00cb80d8',
								service_name: 'consumer-svc',
							},
						},
						{
							data: {
								latency_0: 'n/a',
								latency_1: 502.62,
								latency_2: 'n/a',
								latency_3: 'n/a',
								service_instance_id: '9e87227f-a564-4b55-bf7c-fb00365d9400',
								service_name: 'consumer-svc',
							},
						},
						{
							data: {
								latency_0: 'n/a',
								latency_1: 'n/a',
								latency_2: 502.81,
								latency_3: 'n/a',
								service_instance_id: 'ac4833a8-fbe1-4592-a0ff-241f46a0851d',
								service_name: 'consumer-svc-2',
							},
						},
						{
							data: {
								latency_0: 'n/a',
								latency_1: 'n/a',
								latency_2: 'n/a',
								latency_3: 3230.1,
								service_instance_id: 'a96ff029-6f14-435a-a3d4-ab4742b4347f',
								service_name: 'consumer-svc',
							},
						},
						{
							data: {
								latency_0: 'n/a',
								latency_1: 'n/a',
								latency_2: 'n/a',
								latency_3: 503,
								service_instance_id: 'e33ffd7c-827a-427a-828e-547e00cb80d8',
								service_name: 'consumer-svc',
							},
						},
						{
							data: {
								latency_0: 'n/a',
								latency_1: 'n/a',
								latency_2: 'n/a',
								latency_3: 502.62,
								service_instance_id: '9e87227f-a564-4b55-bf7c-fb00365d9400',
								service_name: 'consumer-svc',
							},
						},
					],
				},
			},
		],
	},
};

export const consumerGrpOptionsPayload = {
	status: 'success',
	data: {
		stringAttributeValues: ['cg4', 'cg1', 'cg2', 'cg3'],
		numberAttributeValues: null,
		boolAttributeValues: null,
	},
};

export const topicsOptionsPayload = {
	status: 'success',
	data: {
		stringAttributeValues: ['topic2', 'topic1', 'topic3', 'topic4'],
		numberAttributeValues: null,
		boolAttributeValues: null,
	},
};

export const partitionsOptionsPayload = {
	status: 'success',
	data: {
		stringAttributeValues: ['1', '2'],
		numberAttributeValues: null,
		boolAttributeValues: null,
	},
};
