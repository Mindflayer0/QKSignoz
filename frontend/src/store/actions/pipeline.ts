import {
	PipelineColumn,
	ProcessorColumn,
} from 'container/PipelinePage/PipelineListsView';
import {
	getEditedDataSource,
	getElementFromArray,
	getRecordIndex,
} from 'container/PipelinePage/PipelineListsView/utils';
import { Dispatch } from 'react';
import AppActions from 'types/actions';
import {
	DELETE_PIPELINE_DATA,
	DELETE_PROCESSOR_DATA,
	NEW_ADD_PIPELINE_DATA,
	NEW_ADD_PROCESSOR_DATA,
	PIPELINE_DATA_ADD,
	PROCESSOR_DATA_ADD,
	UPDATE_PIPELINE_DATA,
	UPDATE_PROCESSOR_DATA,
} from 'types/actions/pipeline';

export const AddDataPipline = (
	pipelineDataAdd: Array<PipelineColumn>,
): ((dispatch: Dispatch<AppActions>) => void) => (
	dispatch: Dispatch<AppActions>,
): void => {
	dispatch({
		type: PIPELINE_DATA_ADD,
		payload: pipelineDataAdd,
	});
};

export const NewAddPiplineData = (
	newAddPiplineData: PipelineColumn,
): ((dispatch: Dispatch<AppActions>) => void) => (
	dispatch: Dispatch<AppActions>,
): void => {
	dispatch({
		type: NEW_ADD_PIPELINE_DATA,
		payload: newAddPiplineData,
	});
};

export const UpdatePipelineData = (
	pipelineData: Array<PipelineColumn>,
	selectedRecord: PipelineColumn | undefined,
	values: PipelineColumn,
	tagsListData: string[] | undefined,
): ((dispatch: Dispatch<AppActions>) => void) => (
	dispatch: Dispatch<AppActions>,
): void => {
	const findRecordIndex = getRecordIndex(
		pipelineData,
		selectedRecord,
		'name' as never,
	);
	const updatedPipelineData = {
		...pipelineData[findRecordIndex],
		name: values.name,
		alias: values.alias,
		filter: values.filter,
		tags: tagsListData,
	};

	const editedPipelineData = getEditedDataSource(
		pipelineData,
		selectedRecord,
		'name' as never,
		updatedPipelineData,
	);

	dispatch({
		type: UPDATE_PIPELINE_DATA,
		payload: editedPipelineData as Array<PipelineColumn>,
	});
};

export const DeletePipelineData = (
	pipelineData: Array<PipelineColumn>,
	record: PipelineColumn,
): ((dispatch: Dispatch<AppActions>) => void) => (
	dispatch: Dispatch<AppActions>,
): void => {
	const findElement = getElementFromArray(pipelineData, record, 'orderid');

	dispatch({
		type: DELETE_PIPELINE_DATA,
		payload: findElement,
	});
};

export const ProcessorDataAdd = (
	processorData: Array<ProcessorColumn>,
): ((dispatch: Dispatch<AppActions>) => void) => (
	dispatch: Dispatch<AppActions>,
): void => {
	dispatch({
		type: PROCESSOR_DATA_ADD,
		payload: processorData,
	});
};

export const NewAddProcessorData = (
	processorData: ProcessorColumn,
): ((dispatch: Dispatch<AppActions>) => void) => (
	dispatch: Dispatch<AppActions>,
): void => {
	dispatch({
		type: NEW_ADD_PROCESSOR_DATA,
		payload: processorData,
	});
};

export const UpdateProcessorData = (
	processorData: Array<ProcessorColumn>,
	selectedProcessorData: ProcessorColumn | undefined,
	values: ProcessorColumn,
	processorType: string,
): ((dispatch: Dispatch<AppActions>) => void) => (
	dispatch: Dispatch<AppActions>,
): void => {
	const findRecordIndex = getRecordIndex(
		processorData,
		selectedProcessorData,
		'processorName' as never,
	);

	const updatedProcessorData = {
		...processorData[findRecordIndex],
		type: processorType,
		processorName: values.processorName,
		description: values.description,
		name: values.processorName,
	};

	const editedProcessorData = getEditedDataSource(
		processorData,
		selectedProcessorData,
		'processorName' as never,
		updatedProcessorData,
	);

	dispatch({
		type: UPDATE_PROCESSOR_DATA,
		payload: editedProcessorData as Array<ProcessorColumn>,
	});
};

export const DeleteProcessorData = (
	processorData: Array<ProcessorColumn>,
	record: ProcessorColumn,
): ((dispatch: Dispatch<AppActions>) => void) => (
	dispatch: Dispatch<AppActions>,
): void => {
	const findElement = getElementFromArray(processorData, record, 'id');

	dispatch({
		type: DELETE_PROCESSOR_DATA,
		payload: findElement,
	});
};
