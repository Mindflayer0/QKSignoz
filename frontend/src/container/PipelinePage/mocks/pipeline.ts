import { PipelineColumn } from '../PipelineListsView/types';

export const configurationVerison = '1.0';

export const pipelineMockData: Array<PipelineColumn> = [
	{
		orderid: 1,
		uuid: '22a588b8-cccc-4a49-94f1-2caa28271315',
		createdAt: '1674205513092137500',
		createdBy: {
			username: 'nitya',
			email: 'nityananda@signoz.io',
		},
		updatedAt: 'Mon Feb 06 2023',
		updatedBy: {
			username: 'nitya',
			email: '',
		},
		version: 'myversion1',
		name: 'Apache common parser',
		alias: 'apachecommonparser',
		enabled: true,
		filter: 'attributes.source == nginx',
		tags: ['server', 'app'],
		operators: [
			{
				id: '1',
				type: 'grok',
				name: 'grok use common',
				output: 'grokcommon',
				pattern: '%{COMMONAPACHELOG}',
				parse_from: 'body',
				parse_to: 'attributes',
			},
			{
				id: '2',
				type: 'move',
				name: 'rename auth',
				output: 'renameauth',
				parse_from: 'attributes.auth',
				parse_to: 'attributes.username',
			},
		],
	},
	{
		orderid: 2,
		uuid: 'f58ce70c-5f41-48bc-949d-a822e2409257',
		version: 'myversion1',
		createdAt: '1674205513092137500',
		createdBy: {
			username: 'nitya',
			email: 'nityananda@signoz.io',
		},
		updatedAt: 'Sun Feb 05 2023',
		updatedBy: {
			username: 'nityananda',
			email: '',
		},
		name: 'myservice trace parser',
		alias: 'myservicetraceparser',
		enabled: true,
		filter: 'attributes.source == myservice',
		tags: ['server', 'app', 'host', 'realse'],
		operators: [
			{
				id: '1',
				type: 'trace_parser',
				name: 'Parse trace details',
				output: 'removeoldxtrace_id',
				trace_id: {
					parse_from: 'attributes.xtrace_id',
				},
				span_id: {
					parse_from: 'attributes.xspan_id',
				},
				trace_flags: {
					parse_from: 'attributes.xtrace_flag',
				},
			},
			{
				id: '2',
				type: 'remove',
				name: 'remove old xtrace_id',
				output: 'removeoldxspan_id',
				field: 'attributes.xtrace_id',
			},
			{
				id: '3',
				type: 'remove',
				name: 'remove old xspan_id',
				output: 'removeoldxtrace_flag',
				field: 'attributes.xspan_id',
			},
			{
				id: '4',
				type: 'remove',
				name: 'remove old xtrace_flag',
				output: 'removeoldxtrace_flag',
				field: 'attributes.xtrace_flag',
			},
		],
	},
];
