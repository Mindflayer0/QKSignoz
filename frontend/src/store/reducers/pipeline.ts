import { pipelineMockData } from 'container/PipelinePage/mocks/pipeline';
import {
	PipelineColumn,
	PipelineOperators,
} from 'container/PipelinePage/PipelineListsView';
import {
	PipelineActions,
	SAVE_PIPELINE_DATA,
	SAVE_PROCESSOR_DATA,
} from 'types/actions/pipeline';

export interface PipelineReducerType {
	pipelineData: Array<PipelineColumn>;
	processorData: Array<PipelineOperators>;
}

const initialState = {
	pipelineData: pipelineMockData,
	processorData: [],
};

export const PipelineReducer = (
	state = initialState,
	action: PipelineActions,
): PipelineReducerType => {
	switch (action.type) {
		case SAVE_PIPELINE_DATA:
			return {
				...state,
				pipelineData: action.payload,
			};

		case SAVE_PROCESSOR_DATA:
			return {
				...state,
				processorData: action.payload,
			};

		default:
			return state;
	}
};
