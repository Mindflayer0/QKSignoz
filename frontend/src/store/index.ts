import { applyMiddleware, compose, createStore } from 'redux';
import type { ThunkMiddleware } from 'redux-thunk';
import thunk from 'redux-thunk';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	// @TODO Add Type for AppActions also
	composeEnhancers(applyMiddleware(thunk as ThunkMiddleware<AppState>)),
);

export default store;
