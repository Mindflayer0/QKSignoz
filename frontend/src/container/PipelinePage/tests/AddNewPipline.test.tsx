import { render } from '@testing-library/react';
import { FormInstance } from 'antd';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import { pipelineData } from '../mocks/pipeline';
import AddNewPipline from '../PipelineListsView/AddNewPipline';

beforeAll(() => {
	Object.defineProperty(window, 'matchMedia', {
		writable: true,
		value: jest.fn().mockImplementation((query) => ({
			matches: false,
			media: query,
			onchange: null,
			addListener: jest.fn(),
			removeListener: jest.fn(),
			addEventListener: jest.fn(),
			removeEventListener: jest.fn(),
			dispatchEvent: jest.fn(),
		})),
	});
});

describe('PipelinePage test', () => {
	it('should render AddNewPipline', () => {
		const ref = React.createRef<FormInstance>();
		const setActionType = jest.fn();
		const handleModalCancelAction = jest.fn();
		const setPipelineDataSource = jest.fn();
		const selectedRecord = pipelineData[0];
		const isActionType = 'add-pipeline';
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<AddNewPipline
							isActionType={isActionType}
							setActionType={setActionType}
							selectedRecord={selectedRecord}
							pipelineDataSource={pipelineData}
							setPipelineDataSource={setPipelineDataSource}
							formRef={ref}
							handleModalCancelAction={handleModalCancelAction}
						/>
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
