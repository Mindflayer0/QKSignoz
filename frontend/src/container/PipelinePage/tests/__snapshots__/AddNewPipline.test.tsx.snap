// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelinePage test should render AddNewPipline 1`] = `<DocumentFragment />`;
